src/macaron/slsa_analyzer/registry.py

```diff
@@ -325,6 +342,165 @@ def _validate_check_relationship(relationship: Any) -> bool:
 
         return False
 
+    def get_transitive_children(self, parent_id: str) -> list[str]:
+        """Return a list of all transitive children for a check.
+
+        Parameters
+        ----------
+        parent_id : str
+            The id of the parent check.
+
+        Returns
+        -------
+        list[str]
+            The list of transitive children check id.
+        """
+        visited = []
+        stack = [parent_id]
+
+        while stack:
+            ele = stack[-1]
+
+            if ele not in visited:
+                visited.append(ele)
+
+                if ele not in self._all_checks_mapping:
+                    raise CheckRegistryError(f"Check {parent_id} does not exist")
+
+                children = self._check_relationships_mapping.get(ele, {})
+
+                for child in children:
+                    if child not in visited:
+                        stack.append(child)
+                    elif child in stack:
+                        # Circular Dependency.
+                        raise CheckCircularDependency("cycle nodes detected")
+            else:
+                stack.pop()
+
+        return visited
+
+    def get_transitive_parents(self, child_id: str) -> list[str]:
+        """Return a list of all transitive parent for a check.
+
+        Parameters
+        ----------
+        child_id : str
+            The id of the child check.
+
+        Returns
+        -------
+        list[str]
+            The list of transitive parents check id.
+        """
+        visited = []
+        stack = [child_id]
+
+        while stack:
+            ele = stack[-1]
+
+            if ele not in visited:
+                visited.append(ele)
+
+                if ele not in self._all_checks_mapping:
+                    raise CheckRegistryError(f"Check {ele} does not exist")
+
+                check = self._all_checks_mapping.get(ele)
+
+                for relation in check.depends_on:  # type: ignore
+                    parent = relation[0]
+                    if parent not in visited:
+                        stack.append(parent)
+                    elif parent in stack:
+                        # Circular Dependency.
+                        raise CheckCircularDependency("cycle nodes detected")
```

Comment:
This Error is only raised when in this exact scenario:
- The user provide custom exclude/include config.
- Their are circular dependencies in the checks.
This Error was implemented because to obtain the transitive parents or children of an exclude/include check, we need to traverse the dependency tree. Therefore, we want to somehow report if there are any circular dependencies. Otherwise, if the user don't use the exclude/include feature, any circular dependencies are caught by the graphlib library [here](https://github.com/oracle/macaron/blob/f6d7426d7b9f937477fbe5ab5cda809edd6dc18b/src/macaron/slsa_analyzer/registry.py#L689).
To print out more details on the circular path, more logic could be add. Do you think it would complicate the method even more :thinking: ?